src/operator.test.js

```diff
@@ -0,0 +1,109 @@
+const { ResourceEventType } = require("@dot-i/k8s-operator");
+const sinon = require("sinon");
+const { expect } = require("./test-setup");
+const Operator = require("./operator");
+const SecretHandler = require("./secret-handler");
+
+describe("operator", () => {
+  beforeEach(() => {
+    this.secretHandlerBackupMock = sinon
+      .stub(SecretHandler.prototype, "backup")
+      .resolves();
+
+    this.operator = new Operator();
+
+    this.watchResourceMock = sinon.stub(this.operator, "watchResource");
+  });
+  afterEach(() => {
+    this.operator.stop();
+    sinon.restore();
+  });
+
+  it("should not call secretHandlerBackupMock if the secret has 'cert-manager.io/certificate-name' annotation", async () => {
+    const invalidTestEvent = {
+      object: {
+        metadata: {
+          annotations: {
+            "cert-manager.io/certificate-name": "tls"
+          }
+        }
+      }
+    };
+
+    await this.watchResourceMock.callsArgWith(3, invalidTestEvent);
+
+    await this.operator.start();
+
+    expect(this.secretHandlerBackupMock).to.not.have.been.called;
+  });
+
+  it("should not call secretHandlerBackupMock if the event type is ResourceEventType.Added", async () => {
```

Comment:
I would generate these not backuping scenarios with a for each. I think that would make it easier to understand and extend, and reduce the repetition amount in these 3 tests.
Also I think it would be better to describe the behaviour of the code with something like `it should not create a backup when the event type is ResourceEventType.Added`